crates/rpc-types/src/eth/transaction/request.rs

```diff
@@ -460,6 +460,18 @@ impl From<TxEnvelope> for TransactionRequest {
     }
 }
 
+/// Error thrown when `to` is `None` in non-EIP4844 transaction requests 
+#[derive(Debug, Default, thiserror::Error)]
+#[error("the field `to` is unset. Please populate it.")]
+#[non_exhaustive]
+pub struct TransactionRequestNon4844Error;
+
+/// Error thrown when `to` is `None` in non-EIP4844 transaction requests 
+#[derive(Debug, Default, thiserror::Error)]
+#[error("the field `to` can only be of type TxKind::Call(Account). Please change it accordingly.")]
+#[non_exhaustive]
+pub struct TransactionRequest4844Error;
+
```

Comment:
so far we have not introduced explicit error types for most tx request fields, we've been using strings for missing keys. So i'd be fine if these structs were omitted